keylime/ima/ima.py

```diff
@@ -197,6 +212,11 @@ def _validate_ima_sig(
             logger.debug("IMA: ignoring excluded path %s", path.name)
             return failure
 
+        if not digest.algorithm:
```

Comment:
I think this we set the `algorithm` always in the `Digest` class. The algorithm is never set in the old `d` template.

Reply:
Then that should probably be done in the Digest class then. The problem is this here:

```
class Digest:
    """
    Class for types "d" and "d-ng" with and without algorithm
    """

    hash: bytes
    algorithm: Optional[str] = None
[...]
```

It must not be Optional...

Reply:
I now added another patch to the front of this series that fixes this issue.

Reply:
Yes, the provided patch fixes that thanks!